src/main/java/org/opentripplanner/middleware/controllers/response/ResponseList.java

```diff
@@ -0,0 +1,60 @@
+package org.opentripplanner.middleware.controllers.response;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Generic class for wrapping a paginated list response for a 'get all' HTTP endpoint.
+ */
+public class ResponseList<T> {
+    public List<T> data;
+    public int offset;
+    public int limit;
+    public long total;
+    public Date timestamp;
+
+    /**
+     * No-arg constructor for de/serialization.
+     */
+    public ResponseList() { }
+
+    /**
+     * Primary constructor for generating a paginated response list.
+     * @param collection - Mongo collection from which to construct the list
+     * @param filter - filter (query/sort) to apply to find operation (null is OK)
+     * @param offset - number of results by which to offset
+     * @param limit - number of results by which the response should be limited
+     */
+    public ResponseList(MongoCollection<T> collection, Bson filter, int offset, int limit){
+        FindIterable<T> iterable = filter != null ? collection.find(filter) : collection.find();
+        this.data = iterable.skip(offset).limit(limit).into(new ArrayList<>());
+        this.offset = offset;
+        this.limit = limit;
+        this.total = collection.countDocuments();
```

Comment:
Wouldn't this count all documents and not just those that the query would filter to? Seems like the `total` should be the total amount of records found in the query.